.github/workflows/ci.yaml

```
@@ -54,12 +55,20 @@ jobs:
       - name: Composer install
         run: |
           docker-compose exec -T web composer config --merge --json "extra.patches" '{"drupal/group": {"Using a translatable string as a category for field type is deprecated - https://www.drupal.org/project/group/issues/3458530": "https://www.drupal.org/files/issues/2024-07-02/group-translate-string-as-category-is-deprecated_0.patch"}}'
-          docker-compose exec -T web composer require --dev cweagans/composer-patches
+          docker-compose exec -T web composer require --dev cweagans/composer-patches drupal/groupmedia:~${{ matrix.groupmedia }}
 
       - name: Drupal site install
         run: |
           docker-compose exec -T web ./vendor/bin/run drupal:site-install
 
       - name: PhpUnit
+        if: ${{ matrix.groupmedia != '3' }}
         run: |
-          docker-compose exec -T web ./vendor/bin/phpunit -vvv --debug
+            docker-compose exec -T web ./vendor/bin/phpunit -vvv --debug
+
+      - name: PhpUnit
+        if: ${{ matrix.groupmedia == '3' }}
+        run: |
+            docker-compose exec -T web ./vendor/bin/phpunit -vvv --debug --testsuite "Collabora Online Group"
+
+
```

Instead of the matrix, we could run the composer require in a second phpunit step:
```suggestion
          # Test with groupmedia 3.
          docker-compose exec -T web ./vendor/bin/phpunit -vvv --debug

      - name: "PhpUnit - groupmedia 4"
        run: |
          docker-compose exec -T web composer require --dev drupal/groupmedia:^4
          docker-compose exec -T web ./vendor/bin/phpunit -vvv --debug --testsuite "Collabora Online Group"
```

In this way we skip a full docker-compose install.

Of course there is a tradeoff here:
- With the matrix, the extra step happens in a separate workflow run, so it would be in parallel, which _should_ finish faster, but could be punished by github (if github slows down projects that have many parallel tasks).
- Without the matrix, the individual workflow run takes a bit more time, but we reduce the number of parallel runs.

In this case the extra composer require + phpunit run takes less time than the initial setup (docker-compose, drupal site-install), so I think this is the way to go.

---

Restored the smoke test since we don't perform full installation on any step, and the other tests are Kernel.

---

Yes, I think is better like this since we don't have to run a full installation and run all the tests, added the extra step.